vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/treegrid/it/TreeGridExpandedKeysPage.java

```diff
@@ -0,0 +1,58 @@
+package com.vaadin.flow.component.treegrid.it;
+
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Input;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.treegrid.TreeGrid;
+import com.vaadin.flow.data.provider.DataKeyMapper;
+import com.vaadin.flow.data.provider.hierarchy.TreeData;
+import com.vaadin.flow.data.provider.hierarchy.TreeDataProvider;
+import com.vaadin.flow.data.value.ValueChangeMode;
+import com.vaadin.flow.router.Route;
+
+import java.util.Arrays;
+
+@Route("vaadin-grid/treegrid-expanded-keys")
+public class TreeGridExpandedKeysPage extends Div {
+
+    public TreeGridExpandedKeysPage() {
+        var grid = new TreeGrid<String>();
+        grid.setPageSize(10);
+        grid.addHierarchyColumn(String::toString).setHeader("Item");
+
+        var data = new TreeData<String>();
```

Comment:
I'm inclined to recommend this test setup, as it's less likely to be affected by future refactorings:

1. Expand All
2. Scroll to 0-250
3. Scroll to start

```java
TreeGrid grid = new TreeGrid<String>();
grid.setPageSize(10);
grid.addHierarchyColumn(String::toString).setHeader("Item");

var data = new TreeData<String>();

for (int i = 0; i < 500; i++) {
    data.addRootItems("Item-" + i);
}

for (int i = 0; i < 500; i++) {
    data.addItems("Item-0", "Item-0-" + i);
}

for (int j = 0; j < 500; j++) {
    data.addItems("Item-0-0", "Item-0-0-" + j);
}

grid.setDataProvider(new TreeDataProvider<>(data));

add(grid);

add(new NativeButton("Expand all",
        e -> grid.expandRecursively(data.getRootItems(), 1)));

add(new NativeButton("Scroll to 0-250",
        e -> grid.scrollToIndex(0, 250)));

add(new NativeButton("Scroll to start",
        e -> grid.scrollToStart()));
```

Reply:
Modified the test as recommended. Also added id checks in every step of the test.